Shop.BLL/Exceptions/NotFoundExceptions/UserNotFoundException.cs

```diff
@@ -0,0 +1,8 @@
+namespace Shop.BLL.Exceptions.NotFoundExceptions;
+
+public class UserNotFoundException : NotFoundException
+{
+    public UserNotFoundException(Guid id) : base($"User with id: {id} not found.")
```

Comment:
create a NotFoundException and pass a string message to the constructor, and raise an exception in the service and pass a specific message there